src/OgHelper.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\og\OgHelper.
+ */
+
+namespace Drupal\og;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Field\FieldException;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+
+/**
+ * OG helper methods.
+ */
+class OgHelper {
+
+  /**
+   * Return TRUE if a field can be used and has not reached maximum values.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The content entity to check the field cardinality for.
+   * @param string $field_name
+   *   The field name to check the cardinality of.
+   *
+   * @return bool
+   */
+  public static function checkFieldCardinality(ContentEntityInterface $entity, $field_name) {
```

Comment:
What would be the (real world) use case for `FieldableEntityInterface`?

Anyway, just wanted to put that question out there - we can revisit it later